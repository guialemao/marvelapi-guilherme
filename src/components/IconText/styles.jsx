import styled from 'styled-components';

export const IconTextWrapper = styled.div`
  margin-left: 15px;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
`;

export const Icon = styled.img`
  width: 17px;
  margin-right: 10px;
`;

export const FilterText = styled.span`
  color: #fa7c7c;
  cursor: pointer;
`;
