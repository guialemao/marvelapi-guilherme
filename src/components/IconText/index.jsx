import React from 'react';
import PropTypes from 'prop-types';

import { IconTextWrapper, Icon, FilterText } from './styles';

const IconText = ({ icon, text, handleClick, 'data-testid': dataTestid }) => (
  <IconTextWrapper data-testid={dataTestid} onClick={handleClick}>
    <Icon src={icon} alt="" />
    <FilterText>{text}</FilterText>
  </IconTextWrapper>
);

IconText.propTypes = {
  'data-testid': PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  handleClick: PropTypes.func.isRequired,
};

export default IconText;
