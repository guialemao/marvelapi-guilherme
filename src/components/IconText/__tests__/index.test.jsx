import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';

import IconText from '../index';

afterEach(cleanup);

const mockComponent = {
  icon: 'noun_Superhero_2227044@3x.77c63509.png',
  text: 'Ordernar por nome - A/Z',
  handleClick: jest.fn(),
  orderedArray: [
    { id: 101134, name: '3-D Man' },
    { id: 1017100, name: 'A-Bomb (HAS)' },
  ],
  reverseOrderedArray: [
    { id: 1017100, name: 'A-Bomb (HAS)' },
    { id: 101134, name: '3-D Man' },
  ],
};
const { icon, text, handleClick } = mockComponent;

const component = (
  <IconText
    data-testid="order-by-name"
    icon={icon}
    text={text}
    handleClick={handleClick}
  />
);

describe('Testing Footer component', () => {
  it('should create a snapshot', () => {
    const { asFragment } = render(component);

    expect(asFragment(component)).toMatchSnapshot();
  });

  it('check if text is correct', () => {
    const { getByText } = render(component);
    expect(getByText('Ordernar por nome - A/Z')).toBeTruthy();
  });

  it('test order of array on click', () => {
    const { getByTestId } = render(component);
    fireEvent.click(getByTestId('order-by-name'));

    expect(handleClick).toHaveBeenCalled();
  });
});
