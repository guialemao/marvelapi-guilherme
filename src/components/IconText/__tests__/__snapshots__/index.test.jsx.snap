// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Footer component should create a snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnylx cfFKSt"
  >
    <img
      alt=""
      class="sc-gtssRu dvgkYR"
      src="noun_Superhero_2227044@3x.77c63509.png"
    />
    <span
      class="sc-dlnjPT hhwbcD"
    >
      Ordernar por nome - A/Z
    </span>
  </div>
</DocumentFragment>
`;
